lms/djangoapps/commerce/management/commands/create_orders_for_old_enterprise_course_enrollment.py

```diff
@@ -0,0 +1,261 @@
+                    non_paid += 1
+                    continue
+                enrollment_payload = {
+                    "lms_user_id": user.id,
+                    "username": username,
+                    "email": user.email,
+                    "date_placed": enrollment.created.isoformat(),
+                    "course_run_key": course_id,
+                    "enterprise_customer_name": enterprise_customer.name,
+                    "enterprise_customer_uuid": str(enterprise_customer.uuid),
+                }
+            except AttributeError as ex:
+                self.stdout.write(u'\t\tskipping enrollment {} due to invalid data. {}'.format(enrollment, ex))
+                invalid += 1
+                continue
+            except CourseEnrollment.DoesNotExist:
+                self.stdout.write(u'\t\tskipping enrollment {} because CourseEnrollment not found'.format(enrollment))
+                invalid += 1
+                continue
+            enrollments_payload.append(enrollment_payload)
+
+        self.stdout.write(u'\t\tFound {count} Paid enrollments to sync'.format(count=len(enrollments_payload)))
+        if not enrollments_payload:
+            return 0, 0, invalid, non_paid
+
+        self.stdout.write(u'\t\tSyncing started...')
+        success, failed = self._create_manual_enrollment_orders(enrollments_payload)
+        self.stdout.write(
+            u'\t\tSuccess: {} , Failed: {}, Invalid:{} , Non-Paid: {}'.format(
+                success, failed, invalid, non_paid
+            )
+        )
+        return success, failed, invalid, non_paid
+
+    def _sync(self, enrollments_queryset, enrollments_count, enrollments_batch_size):
+        """
+            Syncs a single site
+        """
+        self.stdout.write(u'Syncing process started.')
+
+        offset = 0
+        enrollments_queue = []
+        enrollments_query_batch_size = 5000
+        successfully_synced_enrollments = 0
+        failed_to_synced_enrollments = 0
+        invalid_enrollments = 0
+        non_paid_enrollments = 0
+
+        while offset < enrollments_count:
+            is_last_iteration = (offset + enrollments_query_batch_size) >= enrollments_count
+            self.stdout.write(
+                u'\tSyncing enrollments batch from {start} to {end}.'.format(
+                    start=offset, end=offset + enrollments_query_batch_size
+                )
+            )
+            enrollments_queue += self._get_batched_enrollments(
+                enrollments_queryset,
+                offset,
+                enrollments_query_batch_size
+            )
+            while len(enrollments_queue) >= enrollments_batch_size \
+                    or (is_last_iteration and enrollments_queue):  # for last iteration need to empty enrollments_queue
+                enrollments_batch = enrollments_queue[:enrollments_batch_size]
+                del enrollments_queue[:enrollments_batch_size]
```

Comment:
should not we remove batch from queue after syncing with ecommerce?

Reply:
Yes, we can remove them after syncing but as we already have a copy of enrollments in `enrollments_batch` so it is safe to delete it here.